test/simulator_tests/spatial_leaf_subsampler_test.py

```diff
@@ -0,0 +1,249 @@
+import unittest
+
+import networkx as nx
+import numpy as np
+import pandas as pd
+
+from cassiopeia.data.CassiopeiaTree import CassiopeiaTree
+from cassiopeia.simulator.LeafSubsampler import (
+    LeafSubsamplerError, 
+    LeafSubsamplerWarning
+)
+from cassiopeia.simulator.SpatialLeafSubsampler import SpatialLeafSubsampler
+
```

Comment:
nit: need to have two lines between imports and code body. I can make this small change for you.